mne/viz/epochs.py

```diff
@@ -143,88 +380,128 @@ def plot_epochs_image(epochs, picks=None, sigma=0., vmin=None,
+            (3, 10), (0, 0), colspan=9 if colorbar else 10,
+            rowspan=2 if evoked else 3)
+        if evoked:
+            axes_dict["evoked"] = plt.subplot2grid(
+                (3, 10), (2, 0), colspan=9 if colorbar else 10, rowspan=1)
         if colorbar:
-            cbar = plt.colorbar(im, cax=ax3)
-            if cmap[1]:
-                ax1.CB = DraggableColorbar(cbar, im)
-            tight_layout(fig=this_fig)
+            axes_dict["colorbar"] = plt.subplot2grid((3, 10), (0, 9),
+                                                     colspan=1, rowspan=3)
+    else:
+        if fig is not None:
+            raise ValueError('Both figure and axes were passed, please'
+                             'only pass one of these.')
+        from .utils import _validate_if_list_of_axes
+        oblig_len = 3 - ((not colorbar) + (not evoked))
+        _validate_if_list_of_axes(axes, obligatory_len=oblig_len)
+        axes_dict["image"] = axes[0]
+        if evoked:
+            axes_dict["evoked"] = axes[1]
+        # if axes were passed - we ignore fig param and get figure from axes
+        fig = axes_dict["image"].get_figure()
+        if colorbar:
+            axes_dict["colorbar"] = axes[-1]
+    return axes_dict
+
+
+def _plot_epochs_image(epochs, data, ch_type, vmin=None, vmax=None,
+                       colorbar=False, show=False, unit=None, cmap=None,
+                       axes_dict=None, overlay_times=None, title=None,
+                       evoked=False, ts_args=None):
+    """Plot epochs image. Helper function for plot_epochs_image."""
+    if cmap is None:
+        cmap = "Reds" if data.min() >= 0 else 'RdBu_r'
+
+    # Plot
+    # draw the image
+    ax = axes_dict["image"]
+    fig = ax.get_figure()
+    cmap = _setup_cmap(cmap)
+    n_epochs = len(data)
+    extent = [1e3 * epochs.times[0], 1e3 * epochs.times[-1], 0, n_epochs]
+    im = ax.imshow(data, vmin=vmin, vmax=vmax, cmap=cmap[0], aspect='auto',
+                   origin='lower', interpolation='nearest', extent=extent)
+    if overlay_times is not None:
+        ax.plot(1e3 * overlay_times, 0.5 + np.arange(n_epochs), 'k',
+                linewidth=2)
+    ax.set_title(title)
+    ax.set_ylabel('Epochs')
+    ax.axis('auto')
+    ax.axis('tight')
+    ax.axvline(0, color='k', linewidth=1, linestyle='--')
+
+    # draw the evoked
+    if evoked:
+        from mne.viz import plot_compare_evokeds
+        plot_compare_evokeds(
+            {"cond": list(epochs.iter_evoked())}, axes=axes_dict["evoked"],
+            **ts_args)
+        axes_dict["evoked"].set_xlim(epochs.times[[0, -1]])
+        ax.set_xticks(())
+
+    # draw the colorbar
+    if colorbar:
+        import matplotlib.pyplot as plt
+        cbar = plt.colorbar(im, cax=axes_dict['colorbar'])
+        cbar.ax.set_ylabel(unit + "\n\n", rotation=270)
+        if cmap[1]:
+            ax.CB = DraggableColorbar(cbar, im)
+        tight_layout(fig=fig)
+    fig._axes_dict = axes_dict
```

Comment:
This is a monkey patch, right? If so please add a comment about it lest people think you're setting an existing private `matplotlib` var